ui/desktop/app/services/rdp.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Service from '@ember/service';
+import { service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+
+export default class RdpService extends Service {
+  // =services
+
+  @service ipc;
+
+  // =properties
+
+  /**
+   * The preferred RDP client set by the user.
+   * @type {string|null}
+   * @private
+   */
+  @tracked _preferredRdpClient = null;
```

Comment:
Should the default be `none`? It seems like that's what we fall back to even in error scenarios